File: Assessment/Assessment.WEB/src/app/services/studentassessment.service.ts
import { Inject } from '@angular/core';
import { Http } from '@angular/http';
import { StudentAssessment } from '../entities/studentassessment';

export class StudentAssessmentService {
    http: Http;
    baseUrl: string;
    controller: string = "api/StudentAssessment/";

    constructor(http: Http, baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public getAll() {
        return this.http.get(this.baseUrl + this.controller);
    }

    public get(id: number) {
        return this.http.get(this.baseUrl + this.controller + id);
    }

    public update(studentassessment: StudentAssessment) {
        return this.http.put(this.baseUrl + this.controller, studentassessment);
    }

    public insert(studentassessment: StudentAssessment) {
        return this.http.post(this.baseUrl + this.controller, studentassessment);
    }

    public delete(studentassessment: StudentAssessment) {
        return this.http.delete(this.baseUrl + this.controller);
    }
}

File: Assessment/Assessment.Entities/StudentAssessmentService.cs
using System;
using System.Collections.Generic;
using Assessment.Entities.DataInterface;

namespace Assessment.Entities
{
    public class StudentAssessmentService
    {
        private IStudentAssessmentRepository _repository;

        public StudentAssessmentService(IStudentAssessmentRepository repository)
        {
            _repository = repository;
        }

        public void Insert(StudentAssessment studentAssessment)
        {
            _repository.Insert(studentAssessment);
        }

        public void Delete(StudentAssessment studentAssessment)
        {
            _repository.Delete(studentAssessment);
        }

        public void Update(StudentAssessment studentAssessment)
        {
            _repository.Update(studentAssessment);
        }

        public StudentAssessment Get(int studentAssessmentID)
        {
            return _repository.Get(studentAssessmentID);
        }

        public IEnumerable<StudentAssessment> GetAll()
        {
            return _repository.GetAll();
        }
    }
}